sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -339,6 +340,39 @@ object EliminateDistinct extends Rule[LogicalPlan] {
   }
 }
 
+/**
+ * Remove useless FILTER clause for aggregate expressions.
+ * This rule should be applied before RewriteDistinctAggregates.
+ */
+object EliminateAggregateFilter extends Rule[LogicalPlan] {
```

Comment:
Can we put this rule in the main optimizer batch?

Reply:
OK.